lib/internal/streams/operators.js

```diff
@@ -118,27 +144,26 @@ function map(fn, options) {
             continue;
           }
 
-          if (typeof val?.catch === 'function') {
-            val.catch(onDone);
-          }
+          val = PromiseResolve(val);
+          cnt += 1;
+
 
-          queue.push(val);
+          queue.push(PromisePrototypeThen(val, onThen, onCatch));
           if (next) {
             next();
             next = null;
           }
 
-          if (!done && queue.length && queue.length >= concurrency) {
+          if (!done && cnt >= concurrency) {
             await new Promise((resolve) => {
               resume = resolve;
             });
           }
         }
-        queue.push(kEof);
+        queue.push([kEof, null]);
```

Comment:
Not sure what you are doing here?

Reply:
because we want to throw in the `while(true)` (as otherwise we will get an unhandled rejection) we need to mark if got error or not, so the first value is the map `await`ed result and the second value is the error if have